scarlet/display.py

```diff
@@ -149,8 +150,135 @@ def img_to_rgb(img, filter_weights=None, fill_value=0, norm=None):
+
+    fig.tight_layout()
+    plt.close();
+    return fig
+
+def show_sources(sources, observation=None, norm=None, filter_weights=None, show_observed=False, show_rendered=False, show_sed=True, figsize=None):
+
+    if show_observed or show_rendered:
+        assert observation is not None, "Provide matched observation to show observed frame"
+
+    panels = 1 + sum((show_observed, show_rendered, show_sed))
+    if figsize is None:
+        figsize = (3*panels, 3*len(list(sources)))
+    fig, ax = plt.subplots(len(list(sources)), panels, figsize=figsize)
+    for k,src in enumerate(sources):
+
+        if hasattr(src, 'center'):
+            center = np.array(src.center)
+            # center in src bbox coordinates
+            if src.bbox is not None:
+                center_ = center - np.array(src.bbox.origin[1:])
+            else:
+                center_ = center
+            # center in observed coordinates
+            center__ = center - np.array(src.frame.origin[1:])
+        else:
+            center = None
+
+        panel = 0
+        frame_ = src.frame
+        src.set_frame(src.bbox)
+        if isinstance(src, ComponentTree):
+            model = 0
+            seds = []
+            for component in src:
+                model_ = component.get_model()
+                seds.append(model_.sum(axis=(1,2)))
+                model += model_
+        else:
+            model = src.get_model()
+            seds = [model.sum(axis=(1,2))]
+        src.set_frame(frame_)
+        ax[k][panel].imshow(img_to_rgb(model, norm=norm, filter_weights=filter_weights))
+        ax[k][panel].set_title("Model Source {}".format(k))
+        if center is not None:
+            ax[k][panel].plot(*center_[::-1], "wx", mew=1, ms=10)
+
+        if show_rendered:
+            panel += 1
+            model = src.get_model()
+            model = observation.render(model)
+            ax[k][panel].imshow(img_to_rgb(model, norm=norm, filter_weights=filter_weights))
+            ax[k][panel].set_xlim(src.bbox.left, src.bbox.right)
+            ax[k][panel].set_ylim(src.bbox.bottom, src.bbox.top)
+            ax[k][panel].set_title("Model Source {} Rendered".format(k))
+            if center is not None:
+                ax[k][panel].plot(*center__[::-1], "wx", mew=1, ms=10)
+
+        if show_observed:
+            panel += 1
+            ax[k][panel].imshow(img_to_rgb(observation.images, norm=norm, filter_weights=filter_weights))
+            ax[k][panel].set_xlim(src.bbox.left, src.bbox.right)
+            ax[k][panel].set_ylim(src.bbox.bottom, src.bbox.top)
+            ax[k][panel].set_title("Observation".format(k))
+            if center is not None:
+                ax[k][panel].plot(*center__[::-1], "wx", mew=1, ms=10)
+
+        if show_sed:
+            panel += 1
+            for sed in seds:
+                ax[k][panel].plot(sed)
+            ax[k][panel].set_xticks(range(len(sed)))
+            if hasattr(src.frame, 'channels'):
```

Comment:
I don't think that this check works properly. I ran an older version of the quickstart guide that didn't set the frame channels and this crashed with 
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-12-8c708ce7d03a> in <module>
      3                              observation=observation,
      4                              show_rendered=True,
----> 5                              show_observed=True)

~/projects/scarlet/scarlet/display.py in show_sources(sources, observation, norm, filter_weights, show_observed, show_rendered, show_sed, figsize)
    275             ax[k][panel].set_xticks(range(len(sed)))
    276             if hasattr(src.frame, 'channels'):
--> 277                 ax[k][panel].set_xticklabels(src.frame.channels)
    278             ax[k][panel].set_title("SED")
    279             ax[k][panel].set_xlabel("Channel")

~/miniconda3/envs/scarlet/lib/python3.7/site-packages/matplotlib/axes/_base.py in set_xticklabels(self, labels, fontdict, minor, **kwargs)
   3420             kwargs.update(fontdict)
   3421         ret = self.xaxis.set_ticklabels(labels,
-> 3422                                         minor=minor, **kwargs)
   3423         self.stale = True
   3424         return ret

~/miniconda3/envs/scarlet/lib/python3.7/site-packages/matplotlib/axis.py in set_ticklabels(self, ticklabels, minor, *args, **kwargs)
   1661         """
   1662         get_labels = []
-> 1663         for t in ticklabels:
   1664             # try calling get_text() to check whether it is Text object
   1665             # if it is Text, get label content

TypeError: 'NoneType' object is not iterable
```

So I think you also have to check that `src.frame.channels is not None`.